## <span style="color:red">An Exception has occured at cell 5</span>

<a href="https://colab.research.google.com/github/cjhaster/NRSurrogateCatalog/blob/gh-pages/_sources/events/GW200224_222234.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open in Colab"/></a>

In [1]:
%load_ext autoreload
%autoreload 2
%matplotlib inline

import pandas as pd

pd.set_option("display.max_rows", None, "display.max_columns", None)

In [2]:
import pip

try:
    __import__("nrsur_catalog")
except ImportError:
    pip.main(["install", "nrsur_catalog", "-q", ])


# GW200224_222234

Below are some plots for GW200224_222234 from the NRSurrogate Catalog.

In [3]:
from nrsur_catalog import NRsurResult

nrsur_result = NRsurResult.load("GW200224_222234", cache_dir=".nrsur_catalog_cache")
# you can specify a `cache_dir`: folder where data will be downloaded

## Summary

In [4]:
nrsur_result.summary()

Posterior 90% CI  \
Parameter                                                         
$m_1\ [M_{\odot}]$                    ${51.71}_{-3.31}^{+4.29}$   
$m_2\ [M_{\odot}]$                    ${42.96}_{-5.44}^{+3.64}$   
$\mathcal{M}\ [M_{\odot}]$            ${40.73}_{-2.12}^{+1.92}$   
$q$                                    ${0.83}_{-0.15}^{+0.11}$   
$a_1$                                  ${0.37}_{-0.25}^{+0.34}$   
$a_2$                                  ${0.42}_{-0.28}^{+0.35}$   
$\theta_1$                             ${1.25}_{-0.53}^{+0.56}$   
$\theta_2$                             ${1.41}_{-0.57}^{+0.64}$   
$\chi_{\mathrm{eff}}$                  ${0.09}_{-0.09}^{+0.09}$   
$\chi_{\mathrm{p}}$                    ${0.43}_{-0.23}^{+0.27}$   
$\alpha$                               ${3.05}_{-0.03}^{+0.02}$   
$\delta$                              ${-0.16}_{-0.05}^{+0.05}$   
$t_c\ [s]$                    ${1266618172.38}_{-0.00}^{+0.00}$   
$d_L\ [Mpc]$                    ${1683.63}_{-424.16}^{+344.54}$   
$\phi$                                 ${3.00}_{-2.02}^{+2.19}$   
$\phi$                                 ${1.31}_{-0.06}^{+0.05}$   
$\theta$                               ${1.75}_{-0.01}^{+0.01}$   
$\psi$                                 ${1.67}_{-1.45}^{+1.26}$   
$\phi_{\mathrm{JL}}$                   ${2.53}_{-0.97}^{+1.38}$   
$\phi_{12}$                            ${2.79}_{-1.91}^{+2.50}$   
$\theta_{\mathrm{JN}}$                 ${0.63}_{-0.31}^{+0.33}$   
$m_f\ [M_{\odot}]$                    ${89.53}_{-3.72}^{+3.72}$   
$\chi_f$                               ${0.72}_{-0.04}^{+0.04}$   
$v_{\rm kick}\ [{\rm km/s}]$     ${472.72}_{-312.54}^{+613.58}$   

                                                                        Prior  
Parameter                                                                      
$m_1\ [M_{\odot}]$                                                        $-$  
$m_2\ [M_{\odot}]$                                                        $-$  
$\mathcal{M}\ [M_{\odot}]$    $\text{UniformInComponentsChirpMass} [12, 400]$  
$q$                           $\text{UniformInComponentsMassRatio} [0.17, 1]$  
$a_1$                                              $\text{Uniform} [0, 0.99]$  
$a_2$                                              $\text{Uniform} [0, 0.99]$  
$\theta_1$                                            $\text{Sine} [0, 3.14]$  
$\theta_2$                                            $\text{Sine} [0, 3.14]$  
$\chi_{\mathrm{eff}}$                                                     $-$  
$\chi_{\mathrm{p}}$                                                       $-$  
$\alpha$                                                                  $-$  
$\delta$                                                                  $-$  
$t_c\ [s]$                      $\text{Uniform} [1266618172.3, 1266618172.5]$  
$d_L\ [Mpc]$                     $\text{UniformSourceFrame} [100.0, 10000.0]$  
$\phi$                                             $\text{Uniform} [0, 6.28]$  
$\phi$                                             $\text{Uniform} [0, 6.28]$  
$\theta$                                              $\text{Sine} [0, 3.14]$  
$\psi$                                             $\text{Uniform} [0, 3.14]$  
$\phi_{\mathrm{JL}}$                               $\text{Uniform} [0, 6.28]$  
$\phi_{12}$                                        $\text{Uniform} [0, 6.28]$  
$\theta_{\mathrm{JN}}$                                $\text{Sine} [0, 3.14]$  
$m_f\ [M_{\odot}]$                                                        $-$  
$\chi_f$                                                                  $-$  
$v_{\rm kick}\ [{\rm km/s}]$                                              $-$



| Parameter                    | Posterior 90% CI                  | Prior                                           |
|:-----------------------------|:----------------------------------|:------------------------------------------------|
| $m_1\ [M_{\odot}]$           | ${51.71}_{-3.31}^{+4.29}$         | $-$                                             |
| $m_2\ [M_{\odot}]$           | ${42.96}_{-5.44}^{+3.64}$         | $-$                                             |
| $\mathcal{M}\ [M_{\odot}]$   | ${40.73}_{-2.12}^{+1.92}$         | $\text{UniformInComponentsChirpMass} [12, 400]$ |
| $q$                          | ${0.83}_{-0.15}^{+0.11}$          | $\text{UniformInComponentsMassRatio} [0.17, 1]$ |
| $a_1$                        | ${0.37}_{-0.25}^{+0.34}$          | $\text{Uniform} [0, 0.99]$                      |
| $a_2$                        | ${0.42}_{-0.28}^{+0.35}$          | $\text{Uniform} [0, 0.99]$                      |
| $\theta_1$                   | ${1.25}_{-0.53}^{+0.56}$          | $\text{Sine} [0, 3.14]$                         |
| $\theta_2$                   | ${1.41}_{-0.57}^{+0.64}$          | $\text{Sine} [0, 3.14]$                         |
| $\chi_{\mathrm{eff}}$        | ${0.09}_{-0.09}^{+0.09}$          | $-$                                             |
| $\chi_{\mathrm{p}}$          | ${0.43}_{-0.23}^{+0.27}$          | $-$                                             |
| $\alpha$                     | ${3.05}_{-0.03}^{+0.02}$          | $-$                                             |
| $\delta$                     | ${-0.16}_{-0.05}^{+0.05}$         | $-$                                             |
| $t_c\ [s]$                   | ${1266618172.38}_{-0.00}^{+0.00}$ | $\text{Uniform} [1266618172.3, 1266618172.5]$   |
| $d_L\ [Mpc]$                 | ${1683.63}_{-424.16}^{+344.54}$   | $\text{UniformSourceFrame} [100.0, 10000.0]$    |
| $\phi$                       | ${3.00}_{-2.02}^{+2.19}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi$                       | ${1.31}_{-0.06}^{+0.05}$          | $\text{Uniform} [0, 6.28]$                      |
| $\theta$                     | ${1.75}_{-0.01}^{+0.01}$          | $\text{Sine} [0, 3.14]$                         |
| $\psi$                       | ${1.67}_{-1.45}^{+1.26}$          | $\text{Uniform} [0, 3.14]$                      |
| $\phi_{\mathrm{JL}}$         | ${2.53}_{-0.97}^{+1.38}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi_{12}$                  | ${2.79}_{-1.91}^{+2.50}$          | $\text{Uniform} [0, 6.28]$                      |
| $\theta_{\mathrm{JN}}$       | ${0.63}_{-0.31}^{+0.33}$          | $\text{Sine} [0, 3.14]$                         |
| $m_f\ [M_{\odot}]$           | ${89.53}_{-3.72}^{+3.72}$         | $-$                                             |
| $\chi_f$                     | ${0.72}_{-0.04}^{+0.04}$          | $-$                                             |
| $v_{\rm kick}\ [{\rm km/s}]$ | ${472.72}_{-312.54}^{+613.58}$    | $-$                                             |


Lets make some plots!

## <span style="color:red">Ploomber Engine raised an exception due to the cell below </span>

In [5]:
# NRSurrogate corner plots

import os

param_sets = dict(
    mass=["mass_1", "mass_2", "chirp_mass", "mass_ratio"],
    spin=["a_1", "a_2", "tilt_1", "tilt_2"],
    effective_spin=["mass_ratio", "chi_eff", "chi_p"],
    sky_localisation=["luminosity_distance", "ra", "dec"],
    remnant=["final_mass", "final_spin", "final_kick"],
)
for name, params in param_sets.items():
    fname = f"GW200224_222234_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_corner(params)
        fig.savefig(fname)

    if name == "remnant":
        continue

    # LVK-Comparison plots
    fname = f"GW200224_222234_compare_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_lvk_comparison_corner(params)
        fig.savefig(fname)


OSError: Unable to open file (truncated file: eof = 53803008, sblock->base_addr = 0, stored_eof = 337523632)

## Corner Plots

### Mass


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW200224_222234_mass_corner.png"](GW200224_222234_mass_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW200224_222234_compare_mass_corner.png"](GW200224_222234_compare_mass_corner.png)
:::

::::



### Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW200224_222234_spin_corner.png"](GW200224_222234_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW200224_222234_compare_spin_corner.png"](GW200224_222234_compare_spin_corner.png)
:::

::::


### Effective Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW200224_222234_effective_spin_corner.png"](GW200224_222234_effective_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW200224_222234_compare_effective_spin_corner.png"](GW200224_222234_compare_effective_spin_corner.png)
:::

::::



### Sky-localisation


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW200224_222234_sky_localisation_corner.png"](GW200224_222234_sky_localisation_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW200224_222234_compare_sky_localisation_corner.png"](GW200224_222234_compare_sky_localisation_corner.png)
:::

::::



### Remnant

!["GW200224_222234_remnant_corner.png"](GW200224_222234_remnant_corner.png)


## Waveform posterior-predictive plot

In [ ]:
fname = f"GW200224_222234_waveform.png"
if not os.path.isfile(fname):
    fig = nrsur_result.plot_signal(outdir=".")


![waveform](GW200224_222234_waveform.png)

## Animations  
Here are some animations of the posterior.

|Spin       | 
|-----------|
| <iframe width="630" height="315" src="https://www.youtube.com/embed/ZxHoL64dmoM?autoplay=1&mute=1&playlist=ZxHoL64dmoM&loop=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="no-referrer-when-downgrade" allowfullscreen></iframe>    |  

Posterior samples for the dimensionless spin vectors (chi1, chi2) 
of the component black holes. Each purple marker indicates a posterior sample; 
an arrow drawn from the origin to the marker would show the spin vector. 
The outer radii of the spheres correspond to the maximum spin magnitude of 1. 
The spins are shown in the "wave frame" defined at a reference frequency of 20 Hz. 
The x-axis (orange) and y-axis (green) are shown as arrows near the origin; 
the x-y plane is orthogonal to the orbital angular momentum direction. 
The color reflects posterior probability density.

|Remnant       | 
|-----------|
| <iframe width="630" height="315" src="https://www.youtube.com/embed/_HULeQUbrp0?autoplay=1&mute=1&playlist=_HULeQUbrp0&loop=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="no-referrer-when-downgrade" allowfullscreen></iframe>    | 

Posterior samples for the dimensionless spin vector (chif) and the recoil kick velocity 
(vf, units of km/s). Each purple marker indicates a posterior sample; 
an arrow drawn from the origin to the marker would show the spin or kick vector. 
For the spin, the outer radii of the spheres correspond to the maximum spin magnitude of 1. 
For the kick, the outer radius of the sphere corresponds to a kick magnitude of 2500 km/s.
The remnant spin and kick are shown in the "wave frame" defined at a 
reference time of -100 M_det before the peak waveform amplitude,
where M_det is the detector frame total mass.


## Analysis configs
Below are the configs used for the analysis of this job.

In [ ]:
nrsur_result.print_configs()

If you used this data, please [cite this work](../citation.md).

## Comments
Leave a comment in this [issue](https://github.com/nrsur-catalog/NRSurCat-1/issues/new?title=GW200224_222234).